### DISCUSSION TITLE
Visual Studio DevOps benefit

### DISCUSSION THREAD

**Discussion by smajchrz** [AzureCommunityPL/group#164] [Pomoc] [answered]
Cześć!

Pytam bo odpowiedzi nie znalazłem nawet w dokumentacji MS.
Czy posiadając licencje Visual Studio Professional mogę dodać email z tenanta klienta tak aby używać jego Azure DevOps?
I czy taki krok nie narusza żądnej licencji?

https://docs.microsoft.com/en-us/visualstudio/subscriptions/vs-alternate-identity

**Comment by kamil-mrzyglod** [comment, marked as answer]
Najłatwiejszym sposobem by sprawdzić, czy nie naruszasz licencji jest chyba przeczytanie licencji 😉 

Z tego co pamiętam, ADO sprawdza przy zalogowaniu czy masz licencję VS więc jak zalogujesz się używając dodatkowych credentiali to powinno załapać. Zresztą chyba po to jest model z VS abyś mógł używać konta w wielu organizacjach DevOps.

**Reply by smajchrz** [reply]
Działać, działa bo już sprawdziłem ;)

**Reply by wi5nia** [reply]
Tak, dodatkowo jest nawet tak, ze jak masz przypisana platna licencje do MSA w AzDevOps jak Basic czy Basic+Test to jak sie zalogujesz ADO Ci samo przepisze na VS Subscriber

**Reply by JanuszNowak** [reply]
@seb-ast dodatkowo jeżeli masz licencje Visual Studio Enterprise [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/licensing/concurrent-jobs?view=azure-devops&tabs=ms-hosted ) liczy sie to do parallel jobs.
